src/views/Vaults.tsx

```diff
@@ -18,22 +25,73 @@ const Vaults = () => {
     };
   }, [leader, netConfig]);
 
-  const { vaultIdsLoadingError, vaultManagerIds } = useVaultStore();
+  const { managerIdsLoadingError, vaultManagerIds } = useVaultStore();
   const { watchVbankError, brandToInfo } = useAtomValue(appAtom);
 
-  return (
+  const create = (
     <>
-      <h1>Vaults</h1>
-      {vaultIdsLoadingError && <div>{vaultIdsLoadingError}</div>}
-      {watchVbankError && <div>{watchVbankError}</div>}
-      {!vaultIdsLoadingError &&
+      {!managerIdsLoadingError &&
         !watchVbankError &&
         !(vaultManagerIds && brandToInfo) && (
           <div>Loading collateral choices...</div>
         )}
       {vaultManagerIds && brandToInfo && <CollateralChoices />}
     </>
   );
+
+  const manage = <ManageVaults />;
+
+  const buttonText = (() => {
+    switch (mode) {
+      case Modes.create:
+        return 'Back to Manage Vaults';
+      case Modes.manage:
+        return 'Add New Vault';
+      default:
+        return 'Manage Vaults';
+    }
+  })();
+
+  const content = (() => {
+    switch (mode) {
+      case Modes.create:
+        return create;
+      case Modes.manage:
+        return manage;
+      default:
+        return create;
+    }
+  })();
```

Comment:
wouldn't an object map work here?
```suggestion
  const content = modeFns[mode]();
```

Reply:
Yes done